v0/destinations/clickup/util.js

```diff
@@ -0,0 +1,372 @@
+        destinationExternalId.push(extIdObj.id);
+      }
+    });
+  }
+  return destinationExternalId;
+};
+
+/**
+ * Retrieve configured clickup custom fields for the  given list
+ * https://api.clickup.com/api/v2/list/listId/field
+ * ref: https://clickup.com/api
+ * @param {*} listId
+ * @param {*} apiToken
+ * @returns
+ */
+const retrieveCustomFields = async (listId, apiToken) => {
+  const endpoint = getCustomFieldsEndPoint(listId);
+  const requestOptions = {
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: apiToken
+    }
+  };
+  const customFieldsResponse = await httpGET(endpoint, requestOptions);
+  const processedCustomFieldsResponse = processAxiosResponse(
+    customFieldsResponse
+  );
+
+  if (processedCustomFieldsResponse.status !== 200) {
+    throw new CustomError(
+      `[ClickUp]:: Unable to retrieve available custom fields due to ${JSON.stringify(
+        processedCustomFieldsResponse.response
+      )}`,
+      processedCustomFieldsResponse.status
+    );
+  }
+
+  return processedCustomFieldsResponse.response?.fields;
+};
+
+/**
+ * Function to get the custom field value. As clickup is accepting uuid for dropdown/labels, and location in specific format
+ * @param {*} customField
+ * @param {*} value
+ * @returns
+ */
+const findCustomFieldValue = (customField, value) => {
+  switch (customField?.type) {
+    case "drop_down":
+      return getDropDown(customField, value);
+    case "labels":
+      return getLabels(customField, value);
+    case "location":
+      return getLocation(value);
+    default:
+      return value;
+  }
+};
+
+/**
+ * Function to to build the custom field.
+ * returns the array of custom field with each object contains id,value
+ * @param {*} keyToCustomFieldName [{"from":"url","to":"Url"}]
+ * @param {*} properties {"url":"https://www.rudderstack.com/"}
+ * @param {*} listId
+ * @param {*} apiToken
+ * @returns [{"id":"b0f40a94-ea2a-4998-a514-8074d0eddcde","value":"https://www.rudderstack.com/"}]
+ */
+const customFieldsBuilder = async (
+  keyToCustomFieldName,
+  properties,
+  listId,
+  apiToken
+) => {
+  const responseArray = [];
+  // retrieve available clickup custom field for the given list
+  const availableCustomFields = await retrieveCustomFields(listId, apiToken);
+  // convert array to hashMap with key as field name and value as custom field object
+  const availableCustomFieldsMap = getHashFromArrayWithValueAsObject(
+    availableCustomFields,
+    "name",
+    false
+  );
+
+  // convert destination.Config.keyToCustomFieldName to hashMap
+  const configCustomFieldsMap = getHashFromArray(
+    keyToCustomFieldName,
+    "from",
+    "to",
+    false
+  );
+
+  Object.keys(configCustomFieldsMap).forEach(propertiesKey => {
+    let fieldValue = properties[propertiesKey];
+    if (fieldValue) {
+      const fieldName = configCustomFieldsMap[propertiesKey];
+      const customField = availableCustomFieldsMap[fieldName];
```

Comment:
name the variable as `customFieldDetailsObject`

Reply:
Updated